Rdm dev

doerlbh opened the pull request:
Hi guys, I created a vanilla case for RDM class for people who is eager to use. The methods are still pending to be further implemented though, including transforms and dissimilarity computation etc..

doerlbh committed:
[FEAT] Added vanilla RDMs class

doerlbh committed:
[FEAT] Implemented getVector and getMatrix in RDMs class

doerlbh commented:
Added RDMs class and implemented the two main functions:
**get_vectors**()
**get_matrices**()

doerlbh committed:
[DBG] fixed RDMs class initialization

HeikoSchuett committed:
bugfix by=descriptor, descriptor unknown

HeikoSchuett committed:
more bugfix

HeikoSchuett commented:
I shortly went and fixed the unit-test issues. Please have a look through the code climate report and fix those. You can also run these tests with pylint pyrsa locally. 
Otherwise: Great that we now have something to work on.

doerlbh committed:
[DBG] fixed unit issues for Dataset in rdm-dev branch

doerlbh committed:
[DBG] fix test issues in rdm class fm data-dev

[DBG] fix test issues in rdm class fm data-dev

doerlbh committed:
[DBG] more bug fixes in exampledataset fm data-dev

[DBG] more bug fixes in exampledataset fm data-dev

doerlbh committed:
[DBG] more bug fix in neurodataset fm data-dev

[DBG] more bug fix in neurodataset fm data-dev

JasperVanDenBosch commented on pyrsa/util/indicator.py:
```diff
@@ -0,0 +1,62 @@
+    cc = np.unique(c)
+    K = cc.size
+    rows = np.size(c)
+    cc = cc[cc>0]
```
On first look it seems like `identity` and `identity_pos` are quite similar, perhaps  `identity_pos` could call `identity` with an flag `positive=True`? It then checks for this flag and does the `cc[cc>0]` step. This solves the code replication warning.

doerlbh replied:
Thanks for the note! Will take a look and change in the next few pushes.

doerlbh replied:
fixed and pushed.

doerlbh commented:
Will work on the code climate stuff later this week or early next week. Other than that, the unit tests now are all passing with the latest updates from data-dev.

HeikoSchuett committed:
some updates towards pylint style

HeikoSchuett committed:
merge pylint and first implementations

doerlbh committed:
[DBG] style fix on neurodata

doerlbh committed:
[DBG] style fix on rdm-dev

doerlbh committed:
[DBG] more style fix in rdm-dev

doerlbh committed:
[DBG] style fix for data in rdm-dev

doerlbh committed:
[DBG] style fix in rdm

doerlbh committed:
[DBG] style fix in util

doerlbh committed:
[DBG] more style fix in util

doerlbh committed:
[DBG] more style fix in util

doerlbh commented:
@jdiedrichsen @ilogue @HeikoSchuett @iancharest Feel free to checkout the latest version of **rdm-dev**. All tests are passing (unit tests and style tests). It would be great to merge the current stage soon, before more developmental changes are made.

JasperVanDenBosch commented on pyrsa/data/dataset.py:
```diff
@@ -0,0 +1,165 @@
+import pyrsa as rsa
+
+
+class DatasetBase:
```
I think these should go in #16 right?

doerlbh replied:
Yes, that's true, but since I accidentally start my branch rdm-dev from the data-dev, so I have to manually update it in rdm-dev from data-dev, before pull #16 is approved and merged.

JasperVanDenBosch replied:
I'm sorry I don't think I understand. Why's that? Are the dataset classes imported by the rdm modules?

JasperVanDenBosch commented on tests/test_data.py:
```diff
@@ -0,0 +1,30 @@
+import pyrsa.data as rsd
+import numpy as np 
+
+class TestSimulation(unittest.TestCase): 
```
Is this for the Simulation package? It does not seem to test RDM code?

doerlbh replied:
I haven't implemented the test part yet for the rdm classes. I plan to do that as a next step (perhaps in the next pull), or do you suggest me to provide the test now in this pull?

doerlbh replied:
I will add test_rdm.py in my next few pushes. Thanks for the note.

doerlbh replied:
fixed and pushed.

JasperVanDenBosch commented on pyrsa/rdm/__init__.py:
```diff
@@ -0,0 +1 @@
+from pyrsa.data.rdm import RDMs
```
I think this should be `from pyrsa.rdm.rdms import RDMs`

doerlbh replied:
fixed and pushed.

JasperVanDenBosch commented on pyrsa/util/indicator.py:
```diff
@@ -0,0 +1,74 @@
+    indicator_matrix = np.zeros((rows, n_unique))
+    for i in range(n_unique):
+        indicator_matrix[index_vector == c_unique[i], i] = 1
+    return indicator_matrix
```
On first look it seems like `identity` and `identity_pos` are quite similar, perhaps  `identity_pos` could call `identity` with an flag `positive=True`? It then checks for this flag and does the `cc[cc>0]` step. This solves the code replication warning.

doerlbh replied:
Thanks for the note! Will take a look and change in the next few pushes.

doerlbh replied:
fixed and pushed.

JasperVanDenBosch commented on tests/test_indicator.py:
```diff
@@ -0,0 +1,39 @@
+        X = rsu.indicator.allpairs(a)
+        n_row,n_col = X.shape
+        self.assertEqual(n_row,10)
+        self.assertEqual(n_col,5)
```
In the future, it may be useful also to test the contents, e.g. by testing one specific cell. Unless only the shapes of these matrixes matter? I am not very familiar with the concept of indicator matrices.

doerlbh replied:
Thanks for the note! Will take a look and change in the next few pushes. I think the shape is important to check to avoid dimension conflicts, while adding content check will definitely be essential as well.

doerlbh replied:
Fixed and pushed. As suggested, I added a simple content check. For more extensive content checks, I think Joern would be a better person to add them as he first implemented indicator functions and is more familiar to the concept, and might know which cells are more important to add in test cases.

JasperVanDenBosch reviewed (changes requested):
Good work on the style fixes, I have a few more requests, in particular which PR will contain the dataset classes and to reduce the complexity of the indicator module.

doerlbh committed:
[DBG] bug fix in rdm init

doerlbh committed:
[DBG] integrate cleaned up codes from data-dev

doerlbh committed:
[DBG] update unit tests

doerlbh committed:
[DBG] minor bug fix

doerlbh committed:
[DBG] fix test corresponding to new identify init

doerlbh committed:
[DBG] integrate test advice in indicator

doerlbh committed:
[DBG] indicator test fix

doerlbh committed:
[FEAT] add tests for RDM initialization

doerlbh committed:
[DBG] fix RDM init

doerlbh committed:
[FEAT] add scipy requirement

doerlbh committed:
[DBG] fix typo in rdm class

doerlbh committed:
[DBG] fix wrong test for RDM

doerlbh committed:
[FEAT] Added test for get_vectors in RDM

doerlbh committed:
[FEAT] Added test for get_matrices in RDM

doerlbh committed:
[DBG] fix rdm get_matrices

doerlbh committed:
[FEAT] rewrite RDM class structures with util functions

doerlbh committed:
[DBG] minor bug fix

doerlbh committed:
[DBG] fix squareform

doerlbh committed:
[DBG] fix squareform

doerlbh committed:
[DBG] minor style fix

doerlbh commented:
Implementations and unit tests for the class with two main functions are all now complete and passing all criteria. Helpful remarks and requested changes from @ilogue are all fixed in the current version. Please review the code and let me know if anything else need to be changed, and approve if you think it's ready. Thanks!

Commit:
Fixed indicator matrix (allpairs)

Previously allpairs indicator matrix only returned zeros (and not 1 -1
contrasts)

doerlbh committed:
[DBG] fix bug and style issues from the last commit

JasperVanDenBosch commented:
@doerlbh The next step on this I think is to merge master (remove the Data contents from this branch)

doerlbh committed:
[DBG] cleaning up for merging to master

doerlbh commented:
@ilogue Thanks! FYI, I have merged all the updates from the master branch, and they are all cleaned up. Feel free to check it out~

JasperVanDenBosch reviewed (approved):
Looks good!

saarbuckle reviewed (approved):
Good stuff
